lib/ddtrace/contrib/rails/configuration/settings.rb

```diff
@@ -81,6 +81,8 @@ def initialize(options = {})
               Datadog.configuration[:action_view][:template_base_path] = value
             end
           end
+
+          option :log_injection, default: false
```

Comment:
The RFC for this feature is that this is an opt-in, and generally, since we're modifying application output I think it's best practice to have it be opt in. You can see this is the case in languages where it's stable, such as node

`https://datadoghq.dev/dd-trace-js/interfaces/traceroptions.html#loginjection`

That being said after looking closer this will arguably also need to support an env var `DD_LOGS_INJECTION=true|false`. Although our option is framework (rails) specific, I think it probably still applies and would be useful to add, so long as we document that it only applies for rails+certain loggers.

https://docs.datadoghq.com/tracing/connect_logs_and_traces/python/#automatically-inject-trace-and-span-ids

Reply:
I think DD_TRACE_LOGS_INJECTION makes sense here. (I think we are moving towards a `DD_TRACE_` prefix, as of recent PRs you worked on)

This one could be the global one, affecting all frameworks, and we could create a separate one `DD_TRACE_RAILS_LOGS_INJECTION` if needed, but I don't think we should introduced it at this point.